tools/QuietBackgroundProcesses/DevHome.QuietBackgroundProcesses.UI/Views/AnalyticSummaryPopup.xaml.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using DevHome.QuietBackgroundProcesses.UI.ViewModels;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+
+namespace DevHome.QuietBackgroundProcesses.UI.Views;
+
+public sealed partial class AnalyticSummaryPopup : ContentDialog
+{
+    public AnalyticSummaryPopupViewModel ViewModel
+    {
+        get;
+    }
+
+    public AnalyticSummaryPopup(QuietBackgroundProcesses.ProcessPerformanceTable? performanceTable)
+    {
+        ViewModel = new AnalyticSummaryPopupViewModel(performanceTable);
+
+        this.InitializeComponent();
+    }
+
+    private void CloseButton_Click(object sender, RoutedEventArgs e)
+    {
+        this.Hide();
+    }
+
+    private void FilterTextBox_TextChanged(object sender, TextChangedEventArgs e)
```

Comment:
You should make this a RelayCommand on the ViewModel and not have to route it through the View.

Reply:
I'm having trouble finding the role of the view in this MVVM paradigm.  Is it always empty?

Reply:
It seems weird at first, but actually kind of yes -- the more you can get out of the View and into the ViewModel the better. I think of it as the ViewModel has everything the View needs to know, but the View could take all the information and lay it out totally differently. So, if the text of the Filter changes, you'd want to know and react to that no matter how the page is laid out.

In case you haven't seen it before, this is a really good in-depth paper about MVVM and I usually refer to it when I have questions: https://dotnet.microsoft.com/en-us/download/e-book/maui/pdf